datasources/src/commonMain/kotlin/com/mirego/trikot/datasources/extensions/PublisherExtensions.kt

```diff
@@ -0,0 +1,82 @@
+package com.mirego.trikot.datasources.extensions
+
+import com.mirego.trikot.datasources.DataState
+import com.mirego.trikot.streams.reactive.filterIs
+import com.mirego.trikot.streams.reactive.filterNotNull
+import com.mirego.trikot.streams.reactive.just
+import com.mirego.trikot.streams.reactive.map
+import com.mirego.trikot.streams.reactive.onErrorReturn
+import com.mirego.trikot.streams.reactive.processors.MapProcessorBlock
+import com.mirego.trikot.streams.reactive.startWith
+import com.mirego.trikot.streams.reactive.switchMap
+import org.reactivestreams.Publisher
+
+@Suppress("USELESS_CAST", "UNCHECKED_CAST")
+fun <T, E : Throwable> Publisher<T>.mapToDataState(pendingValue: T? = null): Publisher<DataState<T, E>> =
+    map { DataState.Data<T, E>(it) as DataState<T, E> }
+        .onErrorReturn {
+            DataState.Error(it as E)
+        }
+        .startWith(DataState.Pending(pendingValue))
+
+fun <T, R, E : Throwable> Publisher<DataState<T, E>>.mapData(block: MapProcessorBlock<T, R>): Publisher<DataState<R, E>> =
+    map { it.mapData(block) }
+
+@Suppress("UNCHECKED_CAST")
+fun <T, R, E : Throwable> Publisher<DataState<T, E>>.switchMapDataState(
+    getSuccessContinuationPublisher: (data: T) -> Publisher<DataState<R, E>>
+): Publisher<DataState<R, E>> {
+    return switchMap {
```

Comment:
This method seems to do exactly the same as 
```
fun <T, R, E : Throwable> Publisher<DataState<T, E>>.switchMapDataState(
    getSuccessContinuationPublisher: (data: T) -> Publisher<DataState<R, E>>,
    getErrorTransform: ((error: E) -> E)? = null
)
```
Below

Reply:
Of of them has an extra parameter "getErrorTransform"

Reply:
Oh, but it's optional. So yeah, seems useless. removing.

Reply:
Actually if I keep only the version with 2 params it seems to have a hard time inferring types at compile time 🤔 
I'll keep both for now, doesn't really hurt.